runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -54,16 +54,23 @@ bool J9::RecognizedCallTransformer::isInlineable(TR::TreeTop* treetop)
    switch(node->getSymbol()->castToMethodSymbol()->getMandatoryRecognizedMethod())
       {
       case TR::java_lang_Integer_rotateLeft:
-      //case TR::java_lang_Math_abs_I: TODO OMR's iabs evaluator does not fully comply Java Specs
-      //case TR::java_lang_Math_abs_L: TODO OMR's labs evaluator does not fully comply Java Specs
+      case TR::java_lang_Math_abs_I:
+      case TR::java_lang_Math_abs_L:
       case TR::java_lang_Math_abs_F:
       case TR::java_lang_Math_abs_D:
-         return TR::Compiler->target.cpu.isX86();
+         return TR::Compiler->target.cpu.isX86() ||
+            TR::Compiler->target.cpu.isZ();
       case TR::java_lang_Math_max_I:
       case TR::java_lang_Math_min_I:
       case TR::java_lang_Math_max_L:
       case TR::java_lang_Math_min_L:
-         return TR::Compiler->target.cpu.isX86() && !comp()->getOption(TR_DisableMaxMinOptimization);
+         return !comp()->getOption(TR_DisableMaxMinOptimization);
+      case TR::java_lang_Math_max_F:
+      case TR::java_lang_Math_min_F:
+      case TR::java_lang_Math_max_D:
+      case TR::java_lang_Math_min_D:
+         return (TR::Compiler->target.cpu.isARM()) &&
+            !comp()->getOption(TR_DisableMaxMinOptimization);
```

Comment:
This is already set on ARM so we don't need the `isARM` platform check.

Reply:
Since only arm supports transformation of Max and Min calls, without the `isARM` check this would always return false. I intended to leave this code so in the future it could be enabled on ARM. Instead, should the cases for floating point max and min calls just be removed?

Reply:
> Instead, should the cases for floating point max and min calls just be removed?

I think so yes.

Reply:
770bfce